### PR TITLE
fix(get-app): allow ssh urls

### PR DESCRIPTION
**Pull request by gavindsouza**
Closes https://github.com/frappe/bench/issues/993

### CODE REVIEW

**Comment by gavindsouza** [comment]
Tested manually:
-  without explicitly specifying URL (frappe, erpnext apps)
- using HTTP links (http/https)
- using SSH links (git)

via tests:
- using local file paths
- using HTTP links

File: bench/app.py [hunk 1]
```diff
@@ -100,7 +100,9 @@ def remove_from_excluded_apps_txt(app, bench_path='.'):
 
 def get_app(git_url, branch=None, bench_path='.', skip_assets=False, verbose=False, postprocess=True, overwrite=False):
 	if not os.path.exists(git_url):
-		if not check_url(git_url, raise_err=False):
+		if git_url.startswith('git@'):
+			pass
+		elif not check_url(git_url, raise_err=False):
```
Comment (Thunderbottom) [inline bench/app.py:105]:
```suggestion
		if not git_url.startswith('git@') and not check_url(git_url, raise_err=False):
```

Comment (Alchez) [reply bench/app.py:105]: Would it be better to handle it in the `check_url` function, which could just be calling new `check_ssh_url` and `check_https_url` functions?

Comment (Thunderbottom) [reply bench/app.py:105]: sure, that works too

File: bench/app.py [hunk 2]
```diff
@@ -100,7 +105,7 @@ def remove_from_excluded_apps_txt(app, bench_path='.'):
 
 def get_app(git_url, branch=None, bench_path='.', skip_assets=False, verbose=False, postprocess=True, overwrite=False):
 	if not os.path.exists(git_url):
-		if not check_url(git_url, raise_err=False):
```
Comment (Alchez) [inline bench/app.py:103]: Not sure if this is already on your to-do, but we should remove the `check_url` function as well

Comment (gavindsouza) [reply bench/app.py:103]: Not at the moment; There are a few dead functions floating around bench. We'll need a cleanup, we could even remove all the frappe < v5 support :P

Comment (Alchez) [reply bench/app.py:103]: Oh, for sure

**Comment by gavindsouza** [comment]
@Mergifyio backport master

**Pull request merged**